src/Umbraco.Web/PublishedContentQuery.cs

```diff
@@ -386,38 +392,158 @@ public dynamic Search(Examine.SearchCriteria.ISearchCriteria criteria, Examine.P
+                    .Skip(skip) //uses Examine Skip
+                    .ConvertSearchResultToPublishedContent(_contentCache)
+                    .Take(take); //uses Linq Take
+            }
+
+            //create criteria for all fields
+            var allSearchFieldCriteria = SearchAllFields(term, useWildCards, luceneSearcher);
+
+            return TypedSearch(skip, take, out totalRecords, allSearchFieldCriteria, searcher);
+        }
+
+
+        /// <summary>
+        /// Searches content
+        /// </summary>
+        /// <param name="term"></param>
+        /// <param name="useWildCards"></param>
+        /// <param name="searchProvider"></param>
+        /// <returns></returns>
+        public IEnumerable<IPublishedContent> TypedSearch(string term, bool useWildCards = true, string searchProvider = null)
+        {
+            int total;
+            return TypedSearch(0, 0, out total, term, useWildCards, searchProvider);
         }
-
+
         /// <summary>
         /// Searhes content
-        /// </summary>
+        /// </summary>
+        /// <param name="skip"></param>
+        /// <param name="take"></param>
+        /// <param name="totalRecords"></param>
         /// <param name="criteria"></param>
         /// <param name="searchProvider"></param>
         /// <returns></returns>
-        public IEnumerable<IPublishedContent> TypedSearch(Examine.SearchCriteria.ISearchCriteria criteria, Examine.Providers.BaseSearchProvider searchProvider = null)
+        public IEnumerable<IPublishedContent> TypedSearch(int skip, int take, out int totalRecords, ISearchCriteria criteria, BaseSearchProvider searchProvider = null)
         {
-            if (_typedContentQuery != null) return _typedContentQuery.TypedSearch(criteria, searchProvider);
+            if (_typedContentQuery != null) return _typedContentQuery.TypedSearch(skip, take, out totalRecords, criteria, searchProvider);
 
-            var s = Examine.ExamineManager.Instance.DefaultSearchProvider;
+            var s = ExamineManager.Instance.DefaultSearchProvider;
             if (searchProvider != null)
                 s = searchProvider;
 
-            var results = s.Search(criteria);
-            return results.ConvertSearchResultToPublishedContent(_contentCache);
+            //if both are zero, use the simple native Examine API
+            if (skip == 0 && take == 0)
+            {
+                var r = s.Search(criteria);
+                totalRecords = r.TotalItemCount;
+                return r.ConvertSearchResultToPublishedContent(_contentCache);
+            }
+
+            var maxResults = skip + take;
+
+            var results = s.Search(criteria,
+                //don't return more results than we need for the paging
+                //this is the 'trick' - we need to be able to load enough search results to fill
+                //all items to the maxResults
+                maxResults: maxResults);
+
+            totalRecords = results.TotalItemCount;
```

Comment:
@Shazwazza would this be the correct number to return?
As it would return the total number of items you've loaded in to get the paged trick to work.

My assumption would be that this would have been the value of `take`
Thoughts?

Reply:
@warrenbuckley Nope :) You need to read https://shazwazza.com/post/paging-with-examine/

Reply:
OK in your example from your blog post @Shazwazza 
You do the `results.Skip(skip)` first and then do a `.TotalItemCount`

Where here in this PR, you get the TotalItem count first before doing the Skip, which I am unsure would give the correct number of results, if the results has not been skipped with Examine yet before you set the total count items.

Reply:
`TotalItemCount` never changes, doesn't matter if you skip

Reply:
Well all good then. Code reads fine. Just needs some testing/confirmation from Tim